### PR TITLE
Enable localization in mobile wallet

### LINKED ISSUE #562: Enable multilingual support in mobile wallet
TBD

### PR DESCRIPTION
**Pull request by nop33**
Closes #562

### CODE REVIEW

File: apps/mobile-wallet/package.json [hunk 1]
```diff
@@ -87,6 +87,7 @@
     "react-native-background-timer": "^2.4.1",
     "react-native-gesture-handler": "^2.12.1",
     "react-native-get-random-values": "~1.8.0",
+    "react-native-localize": "^3.2.0",
```
Comment (nop33) [inline apps/mobile-wallet/package.json:90]: This package is useful to get the system language for initializing the default language

File: apps/mobile-wallet/src/features/localization/relativeTime.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  M: string
+  MM: string
+  y: string
+  yy: string
+}
+
+// See https://day.js.org/docs/en/customization/customization
+// and https://github.com/iamkun/dayjs/tree/dev/src/locale
+export const shortRelativeTime: Record<string, RelativeTimeEntries> = {
+  en: {
```
Comment (mvaivre) [inline apps/mobile-wallet/src/features/localization/relativeTime.ts:38]: No way to add those in Crowdin, through a dedicated json file?

File: apps/mobile-wallet/src/features/localization/useLocalization.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import i18next from '~/features/localization/i18n'
+import { Language, languageOptions } from '~/features/localization/languages'
+import { systemLanguageMatchFailed, systemLanguageMatchSucceeded } from '~/features/localization/localizationActions'
+import { shortRelativeTime } from '~/features/localization/relativeTime'
+import { useAppDispatch, useAppSelector } from '~/hooks/redux'
+
+dayjs.extend(updateLocale)
+dayjs.extend(relativeTime)
+
+export const useLocalization = () => {
```
Comment (mvaivre) [inline apps/mobile-wallet/src/features/localization/useLocalization.ts:38]: Nit: what about calling this `useInitializeLocalization` for clarity sake, following what we've done already with `useInitializeClient`?

**Review by mvaivre — COMMENTED** [review]
Great great job!

File: apps/mobile-wallet/src/features/localization/useLocalization.ts (hunk 3 above)
Comment (nop33) [reply apps/mobile-wallet/src/features/localization/useLocalization.ts:38]: I would, but it also takes care of changing the language of i18n when the language in the settings changes, so it's not only about initializing.

File: apps/mobile-wallet/src/features/localization/relativeTime.ts (hunk 2 above)
Comment (nop33) [reply apps/mobile-wallet/src/features/localization/relativeTime.ts]: I decided to completely remove it, dayjs provides sane defaults for relative time, no need to add extra config on our side: https://github.com/iamkun/dayjs/tree/dev/src/locale

**Review by mvaivre — APPROVED** [review]
LGTM 🤝

**Pull request merged**